WordPress/gutenberg
# Directly save if only changing current context

## What?
This PR updates the behaviour of the save hub in the site editor so that it only prompts to review changes when there are changes outside the current context. It also adjusts some styling.

## Why?
As part https://github.com/WordPress/gutenberg/issues/50379 we have been exploring how to improve the save dock. This is a proposal to do just that by adjusting behaviour and making some small style changes.

## How?
Checks to see if there is only one record to save and if that matches the current context based on url params.

## Testing Instructions
1. Open up the site editor
2. Go into styles and preview a new style
3. Notice the save button says "Save"
4. Clicking the save button will not trigger the modal and instead save styles directly
5. Make another style change but this time don't save
6. Go back to root, go to "pages", select a page make an edit but don't save
7. Go back to site view and notice it now says "Review x changes..."
8. Click the button and save modal should trigger

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1072756/69420c3d-c7d1-46e4-837a-e3fd7bc3b792
